pkg/graveler/settings/manager.go

```diff
@@ -70,39 +68,64 @@ func (m *Manager) Save(ctx context.Context, repository *graveler.RepositoryRecor
 
 // SaveIf persists the given setting under the given repository and key. Overrides settings key in KV Store.
 // The setting is persisted only if the current version of the setting matches the given checksum.
-func (m *Manager) SaveIf(ctx context.Context, repository *graveler.RepositoryRecord, key string, setting proto.Message, lastKnownChecksum string) error {
+// If lastKnownChecksum is nil, the setting is persisted only if it does not exist.
+func (m *Manager) SaveIf(ctx context.Context, repository *graveler.RepositoryRecord, key string, setting proto.Message, lastKnownChecksum *string) error {
 	logSetting(logging.FromContext(ctx), repository.RepositoryID, key, setting, "saving repository-level setting")
-	decodedChecksum, err := base64.StdEncoding.DecodeString(lastKnownChecksum)
-	if err != nil {
-		return fmt.Errorf("decode checksum: %w", err)
+	valueWithPredicate, err := m.store.Get(ctx, []byte(graveler.RepoPartition(repository)), []byte(graveler.SettingsPath(key)))
```

Comment:
Case `lastKnownChecksum` is nil we don't need to `Get` the current value, we can just return from SetMsg - it will also remove the check if valueWithPredicate is nil later as successful call should return non-nil.